### ISSUE TITLE
The `prefer-add-event-listener` rule should not apply to `onbeforeunload` that returns something

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/eslint-plugin-unicorn#158]
Meaning inline arrow functions or ones with a `return` statement.

This is because returning something changes the behavior:

```js
window.onbeforeunload = () => 'Custom message';

window.onbeforeunload = () => {
	return 'Custom message';
};
```

The above cannot be easily auto-converted to `addEventListener` as it would need to be turned into:

```js
window.addEventListener('beforeunload', event => {
	event.returnValue = 'Custom message';
});
```

Easier to just skip it then.

// @johnwu93

**Comment by johnwu93** [comment]
This looks pretty straightforward. I can fix this along with the other issue mentioned earlier this week. I will get a good amount of them done by the of the week.

I looked at MDN and it looks like there are no other events that similar to `beforeunload`. Namely, events that require you to mutate a property of an event. So, I will just fix this issue for `beforeunload`.

**Comment by bfred-it** [comment]
Could `window.onbeforeunload` part of the rule as well? If someone changes that to `window.addEventListener('beforeunload')` it should not pass 😄

**Comment by sindresorhus** [comment]
> Could window.onbeforeunload be part of the rule as well? If someone changes that to window.addEventListener('beforeunload') it should not pass 😄

Why should it not pass?

**Comment by bfred-it** [comment]
Because it does not work: https://github.com/sindresorhus/refined-github/commit/1d80e1d24e6823a246649928aee249f4e39a824e#commitcomment-27598115

**Comment by futpib** [comment]
What about `window.onbeforeunload = foo;`? Seems like `onbeforeunload` should be reported only when it is assigned a function expression with no `return x` in it.

**Comment by IssuehuntBot** [comment]
@futpib has submitted a pull request. [See it on IssueHunt](https://issuehunt.io/repos/55832243/issues/158)

**Comment by IssuehuntBot** [comment]
@sindresorhus has rewarded $36.00 to @futpib. [See it on IssueHunt](https://issuehunt.io/repos/55832243/issues/158)

- :moneybag: Total deposit: $40.00
- :tada: Repository reward(0%): $0.00
- :wrench: Service fee(10%): $4.00
